src/test/java/com/rybka/controller/ExchangeControllerTest.java

```diff
@@ -0,0 +1,95 @@
+package com.rybka.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.model.CurrencyData;
+import com.rybka.model.ExchangeRequest;
+import com.rybka.model.ExchangeResultData;
+import com.rybka.service.exchange.ExchangeService;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+import java.util.Random;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@WebMvcTest(ExchangeController.class)
+class ExchangeControllerTest {
+    private MockMvc mockMvc;
+    @MockBean
+    private ExchangeService service;
+    private final Random random = new Random();
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void setUp() {
+        var controller = new ExchangeController(service);
+        mockMvc = standaloneSetup(controller).build();
+    }
+
+    @SneakyThrows
+    @Test
+    public void testOnProperGetResponse() {
+        var currencyBase = "usd";
+        var currencyTarget = "eur";
+        var targetUrl = "/exchange/" + currencyBase + "/" + currencyTarget;
+        var currencyData = buildCurrencyData();
+
+        when(service.loadCurrencyOf(anyString().toUpperCase(), anyString().toUpperCase())).thenReturn(currencyData);
+
+        var currencyDataJson = mapToJson(currencyData);
+
+        mockMvc.perform(get(targetUrl))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(currencyDataJson));
+    }
+
+    @SneakyThrows
+    @Test
+    public void testOnProperPostResponse() {
+        var base = "USD";
+        var type = "BUY";
+        var count = 4d;
+        var targetUrl = "/exchange";
```

Comment:
Would be better to have constants class for endpoints and utilize them in the code and tests.

Reply:
fixed